### PR TITLE
#2519 Fix: PulsarOutgoingChannel tracing properties propagation

### PR DESCRIPTION
**Pull request by michalcukierman**

### CODE REVIEW

File: smallrye-reactive-messaging-pulsar/src/main/java/io/smallrye/reactive/messaging/pulsar/PulsarOutgoingChannel.java [hunk 1]
```diff
@@ -166,12 +165,11 @@ private TypedMessageBuilder<T> toMessageBuilder(Message<?> message, Producer<T>
...
-                Map<String, String> properties = new HashMap<>();
-                TracingUtils.traceOutgoing(instrumenter, message, new PulsarTrace.Builder()
-                        .withProperties(properties)
+                PulsarTrace trace = new PulsarTrace.Builder()
                         .withTopic(producer.getTopic())
-                        .build());
-                messageBuilder.properties(properties);
+                        .build();
+                TracingUtils.traceOutgoing(instrumenter, message, trace);
+                messageBuilder.properties(trace.getProperties());
```
Comment (michalcukierman) [inline smallrye-reactive-messaging-pulsar/src/main/java/io/smallrye/reactive/messaging/pulsar/PulsarOutgoingChannel.java:172]: We need to reuse the same instance of a map.

**Comment by ozangunalp** [comment]
@michalcukierman Thank you for this! We better check this in `io.smallrye.reactive.messaging.pulsar.tracing.TracingPropagationTest`.
I suggest we add an assertion in `testFromPulsarToAppToPulsar` to check for "traceparent" property in the received message properties.

**Comment by ozangunalp** [comment]
@michalcukierman Here's my suggestion : https://github.com/smallrye/smallrye-reactive-messaging/commit/fa5099b063ac9a0b9202bf995a5a67011c17ebca

**Comment by michalcukierman** [comment]
Thanks @ozangunalp, thank you for the test I should have add it at first. It looks much better now.
And sorry for the late update, we are going to launch the product next week, so I have a lot on my plate.

**Comment by michalcukierman** [comment]
I've added one more test for messages with PulsarOutgoingMessageMetadata, to cover this change:
https://github.com/smallrye/smallrye-reactive-messaging/pull/2520/files#diff-b9a66ad40b9d795a40fff17754b68b2bf3da86869a99eebc2581100198500a6aR128-R136

**Review by ozangunalp — APPROVED** [review]

**Pull request merged**